src/storage/txn/commands/txn_heart_beat.rs

```diff
@@ -71,10 +73,28 @@ impl<S: Snapshot, L: LockManager> WriteCommand<S, L> for TxnHeartBeat {
 
         let lock = match reader.load_lock(&self.primary_key)? {
             Some(mut lock) if lock.ts == self.start_ts => {
+                let mut updated = false;
+
                 if lock.ttl < self.advise_ttl {
                     lock.ttl = self.advise_ttl;
+                    updated = true;
+                }
+
+                // only for non-async-commit pipelined transactions, we can update the
+                // min_commit_ts
+                if !lock.use_async_commit
+                    && lock.generation > 0
+                    && self.min_commit_ts > 0
+                    && lock.min_commit_ts < self.min_commit_ts.into()
+                {
+                    lock.min_commit_ts = self.min_commit_ts.into();
```

Comment:
What will be if the `lock.min_commit_ts` is not same with `min_commit_ts` in txn cache? I think the `min_commit_ts` txn cache is always less or equal than `lock.min_commit_ts`, am I right?

Reply:
Yes. `min_commit_ts` in the lock should be the source of truth in tikv side. txn_status_cache only caches a "recent enough" `min_commit_ts`, but does not guarantee its consistency with that in the lock.